#! /bin/sh -e
## 19_fix_dec_esp.dpatch by Christoph Hellwig <hch@lst.de>
##
## DP: Adapt dec_esp SCSI driver to use scsi_host_template

# In October 2005, Christoph Hellwig <hch@lst.de> removed the
# Scsi_Host_Template typedef (d0be4a7d29ad0bd3ce2209dd9e46d410b632db59).
# For some reason, this commit didn't end up in linux-mips git with
# the sync of 2.6.15.

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0



--- a/drivers/scsi/dec_esp.c~	2006-01-22 13:14:35.000000000 +0000
+++ b/drivers/scsi/dec_esp.c	2006-01-22 13:19:03.000000000 +0000
@@ -98,7 +98,7 @@
 static irqreturn_t scsi_dma_err_int(int, void *, struct pt_regs *);
 static irqreturn_t scsi_dma_int(int, void *, struct pt_regs *);
 
-int dec_esp_detect(Scsi_Host_Template * tpnt);
+int dec_esp_detect(struct scsi_host_template * tpnt);
 
 static int dec_esp_release(struct Scsi_Host *shost)
 {
@@ -110,7 +110,7 @@
 	return 0;
 }
 
-static Scsi_Host_Template driver_template = {
+static struct scsi_host_template driver_template = {
 	.proc_name		= "dec_esp",
 	.proc_info		= &esp_proc_info,
 	.name			= "NCR53C94",

